src/cryptonote_core/cryptonote_core.cpp

```diff
@@ -1634,6 +1638,40 @@ namespace cryptonote
     return true;
   }
   //-----------------------------------------------------------------------------------------------
+  bool core::check_service_node_time()
+  {
+  
+    crypto::public_key pubkey = m_service_node_list.get_random_pubkey();
+    crypto::x25519_public_key x_pkey{0};
+    m_service_node_list.access_proof(pubkey, [&](auto &proof) { x_pkey = proof.pubkey_x25519; });
+    if (x_pkey) {
+      m_lmq->request(
+          std::to_string(x_pkey),
+          "quorum.timestamp",
+          [&](bool success, std::vector<std::string> data) {
+            m_service_node_list.record_timestamp_participation(pubkey, success);
+            if(success && data.size() == 1){ 
+              try {
+                std::lock_guard<std::mutex> lk(m_sn_timestamp_lock);
+                time_t local_seconds = time(NULL);
+                uint8_t variance = std::abs(std::stoi(data[0])-local_seconds);
```

Comment:
`could variance overflow?`

Reply:
Also there is a `tools::parse_int` function in `common/string_util.h` that wraps the C++17 std::from_chars to make string-to-integer parsing much nicer than the C `stoi`/`atoi`/etc. functions.